### PR TITLE
NIAD-860: Documentation tweaks

### PR DESCRIPTION
**Pull request by bartek-sarul**
## Description

Update documentation

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-860

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [ ] Acceptance Criteria met
- [ ] Commit messages are meaningful
- [ ] Manually tested
- [ ] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [ ] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [ ] Add/update any relevant documentation

### CODE REVIEW

File: OPERATING.md [hunk 1]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
 The adaptor follows usual conventions for containerised solutions and logs to the console. The chosen container 
 orchestration solution must be configured to store and aggregate the adaptor's logs.
 
-## Correlation Id
+## Default log format
+
+Log format is set [here](https://github.com/nhsconnect/integration-adaptor-lab-results/blob/main/src/main/resources/logback.xml#L4) and can be overriden by setting `LAB_RESULTS_LOGGING_FORMAT` environement variable.
```
Comment (andrewwa-kainos) [inline OPERATING.md:113]:
```suggestion
Log format is set [here](https://github.com/nhsconnect/integration-adaptor-lab-results/blob/main/src/main/resources/logback.xml#L4) and can be overridden by setting the `LAB_RESULTS_LOGGING_FORMAT` environment variable.
```

File: OPERATING.md [hunk 2]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
...
+| logger name    | name of the logger that created log entry         |
+| correlation id | (described below)                                 |
+| message        | message of the log entry                          |
+
+`DEBUG` level must be used with caution as it may result in medical data leak
+
+## Correlation Id
 
-For inbound messages the adaptor generates a correlation id for every message (interchange) downloaded from MESH. 
+The adaptor uses a correlation id to make tracing processing of individual messages easier. Adaptor generates a random correlation id for every message (interchange) downloaded from MESH.
```
Comment (andrewwa-kainos) [inline OPERATING.md:133]:
```suggestion
The adaptor uses a correlation id to make it easier to trace the processing of individual messages. It generates a random correlation id for every message (interchange) downloaded from MESH. 
```

File: OPERATING.md [hunk 3]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
...
 Messages published to the adaptor's three AMQP message queues include a CorrelationId header.
 
-# Linking a GP Practice to a Lab Results system
+# NHSACK
 
-The Lab Results Adaptor and Lab Results system communications synchronise through a sequence number mechanism. Linking a GP 
-Practice to a Lab Results system which have never previously exchanged messages requires no additional setup for 
-synchronisation. All the sequences begin at 1, and the adaptor will start them automatically.
+When new Interchanges are received from MESH, they already contain Interchange Sequence Numbers (EDIFACT Interchange Header) and Message Sequence Numbers (EDIFACT Message Header) that were generated by the sender.
+Lab Results Adaptor has no responsibility nor control over those numbers.
```
Comment (andrewwa-kainos) [inline OPERATING.md:144]:
```suggestion
The Lab Results Adaptor has neither responsibility nor control over those numbers.
```

File: OPERATING.md [hunk 4]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
...
 
-In the case that a new market entrant GP System takes over from an incumbent system the new system must pick up the 
-sequences where the incumbent left off. For every GP/Lab Results link established, the incumbent supplier or Lab Results operator 
-must advise the following:
+However, the Adaptor has the ability to produce NHSACK messages that acts as a summary of Interchange processing status if the Interchange Header requests one.
```
Comment (andrewwa-kainos) [inline OPERATING.md:146]:
```suggestion
However, the Adaptor has the ability to produce NHSACK messages that act as summaries of the Interchange processing status when the Interchange Header requests one.
```

File: OPERATING.md [hunk 5]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
...
+In such case an NHSACK message will be created and sent back to the sender of the initial Interchange.
```
Comment (andrewwa-kainos) [inline OPERATING.md:147]:
```suggestion
In such cases an NHSACK message will be created and sent back to the sender of the initial Interchange.
```

File: OPERATING.md [hunk 6]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
...
+Those NHSACK messages are EDIFACT messages and producing them requires sequence numbers to be generated for both whole NHSACK Interchange and NHSACK Messages.
```
Comment (andrewwa-kainos) [inline OPERATING.md:148]:
```suggestion
Those NHSACK messages are EDIFACT messages and producing them requires sequence numbers to be generated for both the whole NHSACK Interchange and the individual NHSACK Messages.
```

File: OPERATING.md [hunk 7]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
...
-    {
-        _id: 'SIS-<GP Link (Trading Partner) Code>-<HA Link (Trading Partner) Code>',
-        sequenceNumber: <Send Interchange Sequence (SIS) number>
-    }
-    
-    {
-        _id: 'SMS-<GP Link (Trading Partner) Code>-<HA Link (Trading Partner) Code>',
-        sequenceNumber: <Send Message Sequence (SMS) number>
-    }
+NHSACK messages are sent back to the Interchange originator using MESH configured mailbox to a designated recipient mailbox. NHS ID Code to recipient mailbox mappings are defined using `LAB_RESULTS_MESH_RECIPIENT_MAILBOX_ID_MAPPINGS` environment variable (check [README](https://github.com/nhsconnect/integration-adaptor-lab-results/blob/main/README.md))
```
Comment (andrewwa-kainos) [inline OPERATING.md:152]:
```suggestion
NHSACK messages are sent back to the Interchange originator using MESH configured mailbox to a designated recipient mailbox. NHS ID Code to recipient mailbox mappings are defined using the `LAB_RESULTS_MESH_RECIPIENT_MAILBOX_ID_MAPPINGS` environment variable (check [README](README.md))
```

**Review by andrewwa-kainos — COMMENTED** [review]
Few suggestions. Should we note somewhere that NHSACKs are not yet implemented?

File: OPERATING.md [hunk 8]
```diff
@@ -108,44 +108,45 @@ three management endpoints:
...
+## Correlation Id
 
-For inbound messages the adaptor generates a correlation id for every message (interchange) downloaded from MESH. 
+The adaptor uses a correlation id to make tracing processing of individual messages easier. Adaptor generates a random correlation id for every message (interchange) downloaded from MESH. 
 An interchange may contain multiple messages, and the adaptor uses the same correlation id for all messages 
 within an interchange.
 
 Some log lines are uncorrelated to a specific message. These logs have an empty `<Correlation Id>` value.
 
 Messages published to the adaptor's three AMQP message queues include a CorrelationId header.
```
Comment (saileshp-kainos) [inline OPERATING.md:139]:
```suggestion
Messages published to the adaptor's three AMQP message queues include a `CorrelationId` header.
```

File: README.md [hunk 9]
```diff
@@ -75,21 +75,15 @@ Configure the MESH API connection using the following environment variables:
...
-organisations' "HA Trading Partner Codes" and their MESH Mailbox IDs are controlled by this variable. Note: A "Destination HA Cipher" 
-can usually be converted into a "HA Link Code" by appending 1 or 01 to create the four-character code. If in doubt consult 
-with the operator of the LAB_RESULTS instance for the correct value.
+(*) 15 digit NHS Sender ID required for successful NHSACK dispatch
 
 The following three variables control how often the adaptor performs a MESH polling cycle. During a polling cycle the 
 adaptor will download and acknowledge up to "the first 500 messages" (a MESH API limit).
 
-Important: If the MESH mailbox uses workflows other than `LAB_RESULTS_REG` and `LAB_RESULTS_RECEP` then these messages must be
+Important: If the MESH mailbox uses workflows other than `PATH_MEDRPT_V3` then these messages must be
```
Comment (n-r-b) [inline README.md:86]: I don't think we've added this workflow yet to the adapter, we're still supporting `LAB_RESULTS_REG` and `LAB_RESULTS_RECEP`

Comment (n-r-b) [reply README.md:86]: Discussed yesterday

File: README.md [hunk 10]
```diff
@@ -1,5 +1,49 @@
-# integration-adaptor-lab-results
-Integration Adaptor to simplify processing of Pathology and Screening results.
+# Lab Results - Pathology and Screening Adaptor
+
+## Adaptor Scope
+
+The main objective of the Lab Results Adaptor is to hide complex legacy standards and instead present a simple and consistent interface aligned to current NHSD national standards. The adaptor removes the requirement for a GP System to handle the complexities of EDIFACT and MESH messaging.
+
+At a high level Lab Results Adaptor exposes a queue from which the GP System can consume medical reports
```
Comment (n-r-b) [inline README.md:7]:
```suggestion
At a high level, the Lab Results Adaptor exposes a queue from which the GP System can consume medical reports.
```

**Review by n-r-b — APPROVED** [review]

File: OPERATING.md [hunk 11]
```diff
@@ -108,44 +108,47 @@ three management endpoints:
...
+
+`DEBUG` level must be used with caution as it may result in medical data leak
 
-The adaptor uses a correlation id to make tracing individual requests easier. All log lines will contain the text 
-`CorrelationId=<Correlation Id>`. For outbound requests the value of `<Correlation Id>` is either declared in the 
-request headers or randomly generated.
+## Correlation Id
 
-For inbound messages the adaptor generates a correlation id for every message (interchange) downloaded from MESH. 
+The adaptor uses a correlation id to make it easier to trace the processing of individual messages easier. It generates a random correlation id for every message (interchange) downloaded from MESH.
```
Comment (andrewwa-kainos) [inline OPERATING.md:133]: "easier" twice
```suggestion
The adaptor uses a correlation id to make it easier to trace the processing of individual messages. It generates a random correlation id for every message (interchange) downloaded from MESH. 
```

File: README.md [hunk 12]
```diff
@@ -1,5 +1,49 @@
...
+- `<example-id>.edifact.dat` - message provided by laboratory
+- `<example-id>.fhir.json` - message translated to FHIR format
+- `<example-id>.nhsack.dat` - confirmation messasge sent back to the laboratory
+
+Pathology examples can be found here //TODO
+Screening examples can be found here //TODO
+
+## Adaptor Architecture
+
+//TODO link to digital.nhs.uk
```
Comment (andrewwa-kainos) [inline README.md:34]: Can this be done? If not, do we have a task in the backlog for it?

**Review by andrewwa-kainos — COMMENTED** [review]
Just another couple of minor points

File: README.md (hunk 12 above)
Comment (bartek-sarul) [reply README.md:34]: https://gpitbjss.atlassian.net/browse/NIAD-993

**Review by andrewwa-kainos — APPROVED** [review]
LGTM

**Review by n-r-b — APPROVED** [review]

**Pull request merged**